corpus/filter.py

```diff
@@ -69,6 +71,45 @@ def run(self):
                 )
 
 
+class FilterSegmentsByRegexJob(Job):
+    def __init__(self, segment_files, filter_regex, invert_match=False):
```

Comment:
Can you move the typing to the function head?

Reply:
Sure, I can do that. I basically adapted the `FilterSegmentsByListJob` which also has the typing like this (like most other jobs in this file). Should we also refactor them or just do it for the job that is added here?

Reply:
If you want I dont think anyone would oppose more type changes :)